### ISSUE TITLE
Skip blobs already present in destination?

### ISSUE THREAD

**Issue by teekay** [LanceMcCarthy/Action-AzureBlobUpload#276]
I would like to specify that the action should not overwrite blobs that are already present in the destination container. Would that be achievable?

**Comment by LanceMcCarthy** [comment]
clean_destination_folder is set to false by default. Check to see if you're setting it to true by accident.

Or, do you mean to not overwrite blobs with the same filename? If yes, what should the duplicate get renamed as?

**Comment by teekay** [comment]
Thank you. Given that I have a blob named "image.jpg" already in the container, I would like to skip the upload - or specify the condition that has to be met for the blob to be overwritten, e.g. if size differs.

**Comment by LanceMcCarthy** [comment]
Good news, I can implement this without a breaking changes.

![image](https://user-images.githubusercontent.com/3520532/182943760-8b1f3da0-8578-4dc2-a26c-bc160a686487.png)

## Notes
I plan on releasing this as `v.2.0.0` for the following reasons:

* Although I do not create any breaking changes, there is still a slight possibility that the behavior will change for some folks, regardless if they have set the new `delete_if_exists` parameter or not.
* I did quite a bit of refactoring to "componentize" the functionality of multiple tasks. So there's some structural changes, even if no major behavior change.

## Example and Prerelease
I will add an extra example in the README, but for now you can take a look at one of my tests to prepare your workflow steps

```yaml
# This previous step uploads a file 'data-composer-0.0.0.exe'
  - name: Single File Test (with '.' Prefix)
    ...

# This step will upload the same file to the same location using
# delete_if_exists: true
  - name: Delete If Exists Test
    uses: LanceMcCarthy/Action-AzureBlobUpload@v2.0.0
    with:
      connection_string: ${{ secrets.AzureBlobConnectionString }}
      container_name: ci-cd
      source_folder: .\TestFiles\build\bin\data-composer-0.0.0.exe
      destination_folder: SingleFileWithPrefixTest
      clean_destination_folder: false  # important to notice
      delete_if_exists: true           # important to notice

```

Take special notice that `clean_destination_folder` is `false` (default is false). This means that the files will still be in the target blob container. By using `delete_if_exists: true`, it will first delete the existing file before uploading the new one, otherwise it will skip it.

I'm not 100% confident this will be the final outcome, even with my successful local and CI tests, so I'll push this out as a beta for a few days. This will give you the chance to use it as well, I'll come back here when it is published as a pre-release and usable in your workflows.

**Comment by LanceMcCarthy** [comment]
the release is available on the GitHub Marketplace https://github.com/LanceMcCarthy/Action-AzureBlobUpload/releases/tag/v2.0.0-beta

You will need to explicitly use the tag `v2.0.0-beta` in your workflow.

For example:

```yaml
  - name: Delete If Exists Test
    uses: LanceMcCarthy/Action-AzureBlobUpload@v2.0.0-beta
    with:
      connection_string: ${{ secrets.AzureBlobConnectionString }}
      container_name: MyAzureBlobContainer
      source_folder: .\MyFilesToUpload\MyFile.txt
      destination_folder: MyDestinationFolder
      delete_if_exists: true
```

If you have any trouble, please open a new Issue so I can investigate and track it separately from this feature Request issue (i.e. it will get a new number that will get tied to the commits, backlog and PRs).

**Comment by teekay** [comment]
Thank you! It seems to be working. I appreciate your work here. ❤️

**Comment by LanceMcCarthy** [comment]
@teekay This is now published as v2.0.0, you can now update your workflow to use:

`uses: LanceMcCarthy/Action-AzureBlobUpload@v2.0.0`

I have also added a major semver tag `v2`, so you can do this to always get the latest 2.x version:

`uses: LanceMcCarthy/Action-AzureBlobUpload@v2`
